documentation-support/src/main/java/tech/picnic/errorprone/documentation/BugPatternExtractor.java

```diff
@@ -91,7 +94,36 @@ private static <T extends AnnotationValue> T doCast(AnnotationValue value, Class
   }
 
   @AutoValue
+  @JsonDeserialize(as = AutoValue_BugPatternExtractor_BugPatternDocumentation.class)
   abstract static class BugPatternDocumentation {
+    static BugPatternDocumentation create(
+        URI source,
+        String fullyQualifiedName,
+        String name,
+        ImmutableList<String> altNames,
+        String link,
+        ImmutableList<String> tags,
+        String summary,
+        String explanation,
+        SeverityLevel severityLevel,
+        boolean canDisable,
+        ImmutableList<String> suppressionAnnotations) {
+      return new AutoValue_BugPatternExtractor_BugPatternDocumentation(
+          source,
+          fullyQualifiedName,
+          name,
+          altNames,
+          link,
+          tags,
+          summary,
+          explanation,
+          severityLevel,
+          canDisable,
+          suppressionAnnotations);
+    }
```

Comment:
This is boilerplate, but Error Prone flags `new AutoValue_BugPatternExtractor_BugPatternDocumentation` (and similar) calls outside the class in which the immutable type is defined.